### PR TITLE
iOS: Update sync table insert tests

### PR DESCRIPTION
**Pull request by phvannor**
Modify some insert tests and fix up some unit test checks

Clean up some warnings from deprecations in iOS 8

### CODE REVIEW

File: sdk/iOS/test/MSJSONSerializerTests.m [hunk 1]
```diff
@@ -232,7 +232,7 @@ -(void)testDataFromItemReturnsDataWithDatesSerialized
     dateParts.hour = 15;
     dateParts.minute = 44;
     dateParts.calendar = [[NSCalendar alloc]
-                          initWithCalendarIdentifier:NSGregorianCalendar];
+                          initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
```
Comment (brettsam) [inline sdk/iOS/test/MSJSONSerializerTests.m]: Curious: Why did you need to change these?

File: sdk/iOS/test/MSSyncTableTests.m [hunk 2]
```diff
@@ -68,41 +68,56 @@ -(void) testInitWithNameAndClient
...
     
-    XCTAssertTrue([self waitForTest:0.1], @"Test timed out.");
+    [self waitForExpectationsWithTimeout:1.0 handler:nil];
+    
+    // Verify the item was in local store
+    NSError *error = nil;
+    NSDictionary *newItem = [offline readTable:TodoTableNoVersion withItemId:itemId orError:&error];
+    XCTAssertNil(error);
+    XCTAssertNotNil(newItem);
+    XCTAssertEqualObjects(newItem[@"text"], item[@"text"]);
```
Comment (brettsam) [inline sdk/iOS/test/MSSyncTableTests.m]: Was there a bug here?

File: sdk/iOS/test/MSJSONSerializerTests.m (hunk 1 above)
Comment (phvannor) [inline sdk/iOS/test/MSJSONSerializerTests.m]: They have been deprecated with iOS 8 (see: https://developer.apple.com/library/ios/releasenotes/General/iOS80APIDiffs/frameworks/Foundation.html)

**Comment by brettsam** [comment]
Looks good.

File: sdk/iOS/test/MSSyncTableTests.m (hunk 2 above)
Comment (phvannor) [inline sdk/iOS/test/MSSyncTableTests.m]: Not for iOS, but it came up in the Android offline review
